sentry_sdk/integrations/falcon.py

```diff
@@ -186,9 +187,15 @@ def sentry_patched_prepare_middleware(
     falcon.api_helpers.prepare_middleware = sentry_patched_prepare_middleware
 
 
-def _is_falcon_http_error(ex):
-    # type: (BaseException) -> bool
-    return isinstance(ex, (falcon.HTTPError, falcon.http_status.HTTPStatus))
+def _exception_leads_to_http_5xx(ex):
+    # type: (Union[Exception, falcon.HTTPError, falcon.http_status.HTTPStatus]) -> bool
+    is_server_error = isinstance(ex, falcon.HTTPError) and (ex.status or "").startswith(
+        "5"
+    )
```

Comment:
This is good, I noticed that even though there's an `HTTPError.code` field, that's not the HTTP status code but some custom user-defined code.

I'm missing here though looking at `HTTPStatus` with a custom code >= 500.
For example, https://falcon.readthedocs.io/en/stable/_modules/falcon/http_status.html#HTTPStatus says `'748 Confounded by Ponies'`, I believe any code that is not in the [0, 499] range should be treated as an error to report to Sentry, perhaps including misuses of the status field, like not starting with a number, what do you think?

Reply:
In the http standard there are only codes from 100-599. The 7xx codes are joke codes. So I think we should be save when we just everything that is in the 5xx class. See also: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status

Reply:
This is the same topic as in `test_http_status`. It is not clear why raising `HTTPStatus(status="500 Internal Server Error")` would, by design, not report to Sentry. We're fixing an "over reporting", I wonder if we're accidentally introducing a case of under-reporting.

Reply:
Falcon is not using HTTPStatus for reporting errors, so I think we are good in not reporting raised HTTPStatus to Sentry.